Fluid/Filters/ColorFilters.cs

```diff
@@ -10,94 +11,245 @@ public static FilterCollection WithColorFilters(this FilterCollection filters)
             {
-                var colorString = String.Empty;
-                if (color.IsEmpty)
+                color = Color.Empty;
+
+                if (String.IsNullOrEmpty(value))
                 {
-                    return colorString;
+                    return false;
                 }
 
-                colorString = "#" + color.R.ToString("X2", null) + color.G.ToString("X2", null) + color.B.ToString("X2", null);
+                if (value[0] == '#')
+                {
+                    string red, blue, green;
+                    switch (value.Length)
+                    {
+                        case 4:
+                            red = Char.ToString(value[1]);
+                            green = Char.ToString(value[2]);
+                            blue = Char.ToString(value[3]);
+                            if (IsHexadecimal(red) && IsHexadecimal(green) && IsHexadecimal(blue))
+                            {
+                                color = Color.FromArgb(Convert.ToInt32(red + red, 16), Convert.ToInt32(green + green, 16), Convert.ToInt32(blue + blue, 16));
+
+                                return true;
+                            }
+
+                            break;
+                        case 7:
+                            red = value.Substring(1, 2);
+                            green = value.Substring(3, 2);
+                            blue = value.Substring(5, 2);
+                            if (IsHexadecimal(red) && IsHexadecimal(green) && IsHexadecimal(blue))
+                            {
+                                color = Color.FromArgb(Convert.ToInt32(red, 16), Convert.ToInt32(green, 16), Convert.ToInt32(blue, 16));
+
+                                return true;
+                            }
+                            
+                            break;
+                    }
+                }
+
+                return false;
 
-                return colorString;
+                static bool IsHexadecimal(string value) => value.All(c => "0123456789abcdefABCDEF".Contains(c));
             }
+
+            public override string ToString() => $"#{R}{G}{B}".ToLower();
+        }
+
+        private struct RgbColor
+        {
+            public static readonly RgbColor Empty = default;
+
+            public RgbColor(Color color) : this(color.R, color.G, color.B)
+            {
+
+            }
+
+            public RgbColor(double red, double green, double blue)
+            {
+                R = red;
+                G = green;
+                B = blue;
+            }
+
+            public double R { get; }
+
+            public double G { get; }
+
+            public double B { get; }
+
+            public static bool TryParse(string value, out Color color)
+            {
+                if (value.StartsWith("rgb(") && value.EndsWith(")"))
+                {
+                    var rgbColor = value.Substring(4, value.IndexOf(")") - 4).Replace(" ", String.Empty).Split(',');
```

Comment:
```suggestion
                    var rgbColor = value.Substring(4, value.IndexOf(")") - 4).Split(new [] { ',', ' '}, StringSplitOptions.RemoveEmptyEntries);
                    
```

Reply:
Also `value.IndexOf(")")` would return an index, whereas Substring will want a length. A solution is just to provide the first argument, and the paren will be the last char, then add `)` in the list of chars to split from.

Reply:
Ya `Substring` want a length, I used `IndexOf` to find the index and substracts the length of `rgb(` which is 4 to get the result, I  will try to use `)` in the split chars

Reply:
Sorry, didn't see `- 4`, don't change it then

Reply:
Or split by adding `(` and then you don't even need substring, as the elements will be `rgb`, `r`, `g`, and `b`. And there can be spaces anywhere, even at the beginning or the end.

Reply:
> Sorry, didn't see - 4, don't change it then

No problem, I changed ;)

Reply:
> Or split by adding ( and then you don't even need substring, as the elements will be rgb, r, g, and b

This will better